### PR TITLE
[IMP] website: introduce `s_intro_pill` snippet

### PR DESCRIPTION
**Pull request by chgo-odoo**
This PR introduces the new `s_intro_pill` snippet.
- requires : https://github.com/odoo/design-themes/pull/882

image

task-4077581
Part of task-4077427

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/views/snippets/s_experience.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="s_experience" name="Experience">
+    <section class="s_experience pt104 pb104">
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="11">
+                <div class="o_grid_item g-height-8 g-col-lg-8 col-lg-6 align-content-center pt96 pb96" data-name="Box" style="z-index: 2; grid-area: 1 / 3 / 10 / 11; text-align: center;">
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_experience.xml:8]: Following website's conventions for grid items:
- Order element in the DOM according to the desktop visual result (in this case: image block, text block, image block)
- Adapt `z-index` values according to the element order on desktop
- If something needs to change position on mobile, add `order: [x];` inline style + the `order-lg-0` class

File: addons/website/views/snippets/s_experience.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    />
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-2 g-height-9 d-none d-lg-block col-lg-3 pt96 pb96" style="z-index: 1; grid-area: 2 / 8 / 12 / 13;">
+                    <img
+                    class="img-fluid mx-auto"
+                    data-shape="web_editor/geometric_round/geo_round_pill"
+                    data-file-name="s_experience_default_image_2.jpg"
+                    data-original-mimetype="image/jpeg/"
+                    src="/web_editor/image_shape/website.s_experience_default_image_2/web_editor/geometric_round/geo_round_pill.svg"
+                    data-shape-colors=";;;;"/>
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_experience.xml:33]: Defining `data-shape-colors` seems not necessary, do you confirm? 
(same for the other image)

File: addons/website/views/snippets/s_experience.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                    src="/web_editor/image_shape/website.s_experience_default_image/web_editor/geometric_round/geo_round_pill.svg"
+                    data-shape-colors=";;;;"
+                    />
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-2 g-height-9 d-none d-lg-block col-lg-3 pt96 pb96" style="z-index: 1; grid-area: 2 / 8 / 12 / 13;">
+                    <img
+                    class="img-fluid mx-auto"
+                    data-shape="web_editor/geometric_round/geo_round_pill"
+                    data-file-name="s_experience_default_image_2.jpg"
+                    data-original-mimetype="image/jpeg/"
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_experience.xml:31]: 😉 (same for the other image)
```suggestion
                    data-original-mimetype="image/jpeg"
```

File: addons/website/views/snippets/s_experience.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                </div>
+                <div class="o_grid_item oe_img_bg o_grid_item_image g-col-lg-4 g-height-10 col-lg-3 order-first pt96 pb96" style="z-index: 1; grid-area: 1 / 1 / 10 / 6;">
+                    <img
+                    class="img-fluid mx-auto"
+                    data-shape="web_editor/geometric_round/geo_round_pill"
+                    data-file-name="s_experience_default_image.jpg"
+                    data-original-mimetype="image/jpeg/"
+                    src="/web_editor/image_shape/website.s_experience_default_image/web_editor/geometric_round/geo_round_pill.svg"
+                    data-shape-colors=";;;;"
+                    />
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_experience.xml:24]: Everything in one line 😉
(other images as well)

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Thanks @chgo-odoo , some nitpicking remarks for you :)

File: addons/website/views/snippets/s_experience.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                </div>
+                <div class="o_grid_item g-height-8 g-col-lg-8 col-lg-6 align-content-center pt96 pb96" data-name="Box" style="z-index: 2; grid-area: 1 / 3 / 10 / 11; text-align: center;">
+                    <h1 class="display-3">Experience the<br/>best quality services</h1>
+                    <p>
+                        <br/>
+                        <br/>
+                        <a t-att-href="cta_btn_href" class="btn btn-lg btn-primary">Discover our services</a>
+                    </p>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-2 g-height-9 d-none d-lg-block col-lg-3 pt96 pb96" style="z-index: 3; grid-area: 2 / 8 / 12 / 13;">
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_experience.xml:20]: `g-col-[x]` classes and/or `grid-area` rules are not set correclty

https://github.com/user-attachments/assets/fea43f53-a4cd-4fa9-bb61-3951258167f1

File: addons/website/views/snippets/snippets.xml [hunk 6]
```diff
@@ -52,7 +52,9 @@
                 <t t-snippet="website.s_carousel" string="Carousel" group="intro">
                     <keywords>gallery, carousel</keywords>
                 </t>
-
```
Comment (stefanorigano) [inline addons/website/views/snippets/snippets.xml:55]: Do not remove this empty line, it makes sense ;)

File: addons/website/views/snippets/s_experience.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+                <div class="o_grid_item oe_img_bg o_grid_item_image g-col-lg-4 g-height-10 col-lg-3 order-lg-0 pt96 pb96" style="order: first; z-index: 1; grid-area: 1 / 1 / 10 / 6;">
+                    <img class="img-fluid mx-auto" data-shape="web_editor/geometric_round/geo_round_pill" data-file-name="s_experience_default_image.jpg"
+                    data-original-mimetype="image/jpeg" src="/web_editor/image_shape/website.s_experience_default_image/web_editor/geometric_round/geo_round_pill.svg"/>
+                </div>
+                <div class="o_grid_item g-height-8 g-col-lg-8 col-lg-6 align-content-center pt96 pb96" data-name="Box" style="z-index: 2; grid-area: 1 / 3 / 10 / 11; text-align: center;">
+                    <h1 class="display-3">Experience the<br/>best quality services</h1>
+                    <p>
+                        <br/>
+                        <br/>
+                        <a t-att-href="cta_btn_href" class="btn btn-lg btn-primary">Discover our services</a>
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_experience.xml:17]: Shorter and less specific ;)

```suggestion
                        <a t-att-href="cta_btn_href" class="btn btn-lg btn-primary">Discover</a>
```

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Yet some other minors, thanks @chgo-odoo 👍

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Here some other remarks @chgo-odoo , thanks 👍

**Comment by anso-odoo** [comment]
Pushed some changes to: 
- Adapt g-height classes that were not matching with grid-area
- Align g-col-lg and col-lg classes
- Remove ordering classes as it looks like it's not useful

Passing this PR as ready for review. Will resolve conflict when it is approved by website team.

File: addons/website/views/snippets/s_intro_pill.xml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                        data-select-class=""
+                        data-img="/website/static/src/img/snippets_options/align_solo_top.svg"/>
+                <we-button title="Align Middle"
+                        data-select-class="justify-content-center"
+                        data-img="/website/static/src/img/snippets_options/align_solo_middle.svg"/>
+                <we-button title="Align Bottom"
+                        data-select-class="justify-content-end"
+                        data-img="/website/static/src/img/snippets_options/align_solo_bottom.svg"/>
+            </we-button-group>
+        </div>
```
Comment (bso-odoo) [inline addons/website/views/snippets/s_intro_pill.xml:44]: If you toggle the block to Column mode, this option remains displayed while it has no more effect.
You can disable it by implementing `_computeWidgetVisibility` as described in that comment <https://github.com/odoo/odoo/pull/175746#discussion_r1715646403>.

Comment (anso-odoo) [reply addons/website/views/snippets/s_intro_pill.xml]: I changed a bit the implementation. We use now grid padding to center the element so we don't have to add extra scss, options, js etc...

The visual result is basically the same except on mobile where I changed the order of the elements so we can have the title in the viewport

|  | Before | After |
|--------|--------|--------|
| Desktop | Screenshot 2024-08-30 at 12 34 13 | Screenshot 2024-08-30 at 12 32 56 |
| Mobile | Screenshot 2024-08-30 at 12 34 35 | Screenshot 2024-08-30 at 12 33 21 |

File: addons/website/views/snippets/s_intro_pill.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+<template id="s_intro_pill" name="Intro Pill">
+    <section class="s_intro_pill pt40 pb40">
+        <div class="container-fluid">
+            <div class="row o_grid_mode" data-row-count="11">
+                <div class="o_grid_item oe_img_bg o_grid_item_image g-col-lg-5 g-height-9 col-lg-5 order-lg-0" style="order: 2; z-index: 1; grid-area: 1 / 1 / 10 / 6; --grid-item-padding-y: 0px; --grid-item-padding-x: 0px;">
+                    <img class="img-fluid mx-auto" data-shape="web_editor/geometric_round/geo_round_pill" data-file-name="s_intro_pill_default_image.jpg"
+                    data-original-mimetype="image/jpeg" src="/web_editor/image_shape/website.s_intro_pill_default_image/web_editor/geometric_round/geo_round_pill.svg"/>
+                </div>
+                <div class="o_grid_item g-height-11 g-col-lg-8 col-lg-8 order-lg-0" data-name="Box" style="order: 1; z-index: 3; grid-area: 1 / 3 / 12 / 11; --grid-item-padding-y: 120px; text-align: center;">
```
Comment (bso-odoo) [inline addons/website/views/snippets/s_intro_pill.xml:12]: When I try to recreate it with the editor, the `text-align` is on `h1` and `p`.

Comment (anso-odoo) [reply addons/website/views/snippets/s_intro_pill.xml:12]: Indeed, fixed 👍

**Comment by bso-odoo** [comment]
@robodoo delegate=bvr-odoo

**Comment by bvr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**